### DISCUSSION TITLE
Adding date variable to commit message

### DISCUSSION THREAD

**Discussion by fredericharnois** [community/community#26915] [Actions]
<p>Hello!</p>
<p>I’m trying to have yesterday’s date dynamically inserted in commit messages when my action runs.</p>
<p>I’ve gotten to this so far:</p>
<pre><code class="lang-auto">jobs:
  updateData:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    
    - uses: actions/setup-python@v2
      with:
        python-version: '3.7'
        
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install setuptools&gt;=42.0.0
        pip install -r requirements.txt
    - name: Generate file
      run: python app.py

    - name: Get yesterday's date
      id: ydate
      run: echo "set-output name=ydate::$(date -d "yesterday 22:0" +'%Y-%m-%d')"
    
    - name: Update data
      run: |
        git config --global user.name 'username'
        git config --global user.email 'email'
        git add *.csv
        git commit -am 'Added data for ${{ steps.ydate.outputs.ydate }}'
        git push
</code></pre>
<p>The <em>Get yesterday’s date</em> step runs properly and outputs the date but the commit message ends up being 'Added data for '.</p>
<p>How does one integrate/concatenate the output with the rest of the commit message string?</p>
<p>Thanks!</p>

**Comment by kingthorin** [comment]
<p>I wonder if the problem might be the apostrophe s in the step name? Maybe that’s not being parsed well since it isn’t quoted?</p>
<p>Haven’t tested, but plausible.</p>
<p>Try echoing the value before the commit?  (Just as a troubleshooting idea).</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/fredericharnois">@fredericharnois</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>This is due to you lost “::” for <code>set-output</code> command, fixed as below:</p>
<pre><code class="lang-auto">    - name: Get yesterday's date
      id: ydate
      run: |
        echo "::set-output name=ydate::$(date -d "yesterday 22:0" +'%Y-%m-%d')"  # add :: before set-ouput
</code></pre>
<p>Thanks</p>

**Comment by kingthorin** [comment]
<p>Nice catch, I totally overlooked that. :man_facepalming:</p>

**Comment by fredericharnois** [comment]
<p>Thank you so much to you both <a class="mention" href="/weide-zhou">@weide-zhou</a> <a class="mention" href="/kingthorin">@kingthorin</a>! It really was just that.</p>
